src/engine/loopingcontrol.cpp

```diff
@@ -835,27 +923,104 @@ void LoopingControl::slotBeatLoop(double beats, bool keepStartPoint) {
         } else {
             newloopSamples.end += 2;
         }
-    } else if (newloopSamples.end > samples) {
+    }
+    if (newloopSamples.end > samples) {
         // Do not allow beat loops to go beyond the end of the track
         newloopSamples.end = samples;
```

Comment:
I noticed slotLoopScale which is used for resizing manual loops does not let loops get resized if they would go beyond the end of the track. I think this is a better solution. I'll make slotBeatLoop do that.